File: app/src/main/java/de/bornholdtlee/defaultproject/injection/modules/ApplicationModule.java
package de.bornholdtlee.defaultproject.injection.modules;


import dagger.Module;
import dagger.Provides;
import de.bornholdtlee.defaultproject.DefaultApplication;

@Module
public class ApplicationModule {

    private final DefaultApplication application;

    public ApplicationModule(DefaultApplication application) {
        this.application = application;
    }

    @Provides
    @ApplicationScope
    public DefaultApplication provideApplication() {
        return application;
    }

}

File: app/src/main/java/de/bornholdtlee/defaultproject/injection/modules/ApplicationModule.kt
package de.bornholdtlee.defaultproject.injection.modules


import dagger.Module
import dagger.Provides
import de.bornholdtlee.defaultproject.base.BaseApplication
import javax.inject.Singleton

@Module
class ApplicationModule(private val application: BaseApplication) {

    @Provides
    @Singleton
     fun provideApplication(): BaseApplication {
        return application
    }

}
